notifier.cpp

```diff
@@ -46,11 +46,21 @@ static void sigusr1_handler(int signum) {
     print = !print;
 }
 
+static void sig_int_term_handler(int signum) {
+    if (signum == 2)
+        std::cout << "Interrupt signal (" << signum << ") received." << std::endl;
+    else if (signum == 15)
+        std::cout << "Termination signal (" << signum << ") received." << std::endl;
+    exit(signum);
```

Comment:
Running out in this place is too rude. You cannot know where the signal will come and what sequence of actions it will interrupt. It is much better to mark the completion `done`-flag and continue further. The processing cycle is interrupted when the action is completed.